src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java

```diff
@@ -67,6 +69,15 @@ public AppiumElementLocator(SearchContext searchContext, By by, boolean shouldCa
         this.exceptionMessageIfElementNotFound =  "Can't locate an element by this strategy: " + by.toString();
     }
 
+    private static By getBy(By currentBy, SearchContext currentContent) {
+        if (!ContentMappedBy.class.isAssignableFrom(currentBy.getClass())) {
+            return currentBy;
```

Comment:
it might be easier to understand the code if this condition had an explanation comment